leetcode/medium/173._Binary_Search_Tree_Iterator.py

```diff
@@ -0,0 +1,43 @@
+# Link to problem: https://leetcode.com/problems/binary-search-tree-iterator/
+# Loom explaining solution you wrote (5m): TODO
+# Solution:
+# Description
+# Key ideas
+# This is the same as iterative in-order processing
+# The best way to solve this question is to draw serval BSTs and reason through them.
+# Difficulty: Medium
+# Code
+
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, val=0, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+class BSTIterator:
+
+    def _toLeftMostSubtree(self):
+        while self.current_location.left is not None:
+            self.current_depth.append(self.current_location)
+            self.current_location = self.current_location.left
+
+    def __init__(self, root: Optional[TreeNode]):
+        self.current_depth = []
```

Comment:
This works. Essentially the `current_depth` keeps track of function call stack, but instead you're using an actual stack. 

Another more straightforward approach is to simply do an inorder traversal at the start, store all of the nodes in a list, then remove from that list when `.next()` is called.

Reply:
yes that would make sense as well!